src/runtime/String.cpp

```diff
@@ -804,6 +804,77 @@ void* UTF16String::operator new(size_t size)
     return GC_MALLOC_EXPLICITLY_TYPED(size, descr);
 }
 
+String* String::getSubstitution(ExecutionState& state, String* matched, String* str, size_t position, std::vector<String*, GCUtil::gc_malloc_ignore_off_page_allocator<String*>> captures, String* replacement)
+{
```

Comment:
It's a minor suggestion. How about passing a reference of std::vector(captures) instead of copying the vector?